packages/ui/src/pages/AdminUsersPage/AdminUsersPage.vue

```diff
@@ -1,30 +1,238 @@
 <script setup lang="ts">
-import { ref } from "vue";
+import { computed, ref } from "vue";
 import { useRouter } from "vue-router";
 import { useAuthStore } from "@/stores/auth";
 
 import adminAPI from "@/api/admin";
 import AppButton from "@/components/ui/AppButton.vue";
-import { useMutation } from "@tanstack/vue-query";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/vue-query";
+import { ElButton, ElCard, ElNotification } from "element-plus";
 
 const router = useRouter();
 const authStore = useAuthStore();
+const queryClient = useQueryClient();
 
-const name = ref("");
-const promoteUser = useMutation(adminAPI.promoteUserToAdmin, {});
+const usersToAdd = ref<string[]>([]);
+const searchQuery = ref("");
+const { data: adminUsers, isLoading } = useQuery(
+  ["admins"],
+  adminAPI.getAdminUsers
+);
+const { data: allUsers } = useQuery(["users"], adminAPI.getAllUsers);
+const addAdmin = useMutation(
+  (userId: string) => adminAPI.addUserAsAdmin(userId),
+  {
+    onSuccess: () => {
```

Comment:
Is there any error handling? Do you guys use toasts anywhere?

Reply:
We're using notifications from element plus and I will add error notifications to adding/removing admins